### PR TITLE
Support `PathLike`

### LINKED ISSUE #2178: Support validating `os.PathLike` typed variables
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this feature/change is needed
* [x] After submitting this, I commit to one of:
    * Look through open issues and helped at least one other person
    * Hit the "watch" button on this repo to receive notifications and I commit to help at least 2 people that ask questions in the future
    * Implement a Pull Request for a confirmed bug

# Feature Request

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.7.2
            pydantic compiled: True
                 install path: /home/hong/src/pydantic/pydantic
               python version: 3.7.3 (default, Jul 25 2020, 13:03:44)  [GCC 8.3.0]
                     platform: Linux-5.8.0-0.bpo.2-amd64-x86_64-with-debian-10.6
     optional deps. installed: ['typing-extensions', 'email-validator', 'devtools']

```

```py
import os
from pydantic.dataclasses import dataclass

@dataclass
class Data:
    a: os.PathLike
```

Running this script leads to:

```
Traceback (most recent call last):
  File "m.py", line 4, in <module>
    @dataclass
  File "pydantic/dataclasses.py", line 229, in pydantic.dataclasses.dataclass
    return wrap(_cls)
  File "pydantic/dataclasses.py", line 224, in pydantic.dataclasses.dataclass.wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen, config)
  File "pydantic/dataclasses.py", line 160, in pydantic.dataclasses._process_class
    cls.__pydantic_model__ = create_model(
  File "pydantic/main.py", line 931, in pydantic.main.create_model
    return type(__model_name, (__base__,), namespace)
  File "pydantic/main.py", line 262, in pydantic.main.ModelMetaclass.__new__
    fields[ann_name] = inferred = ModelField.infer(
  File "pydantic/fields.py", line 315, in pydantic.fields.ModelField.infer
    return cls(
  File "pydantic/fields.py", line 284, in pydantic.fields.ModelField.__init__
    self.prepare()
  File "pydantic/fields.py", line 361, in pydantic.fields.ModelField.prepare
    self.populate_validators()
  File "pydantic/fields.py", line 542, in pydantic.fields.ModelField.populate_validators
    *(get_validators() if get_validators else list(find_validators(self.type_, self.model_config))),
  File "pydantic/validators.py", line 638, in find_validators
    raise RuntimeError(f'no validator found for {type_}, see `arbitrary_types_allowed` in Config')
RuntimeError: no validator found for <class 'os.PathLike'>, see `arbitrary_types_allowed` in Config
```

It would be good to support this builtin annotation type ([os.PathLike](https://docs.python.org/3/library/os.html#os.PathLike)).

### PR DESCRIPTION
**Pull request by samuelcolvin**
fix #2178

Selected Reviewer: @hramezani

### CODE REVIEW

**Comment by samuelcolvin** [comment]
please review.

File: pydantic/_internal/_std_types_schema.py [hunk 1]
```diff
@@ -190,31 +191,31 @@ def uuid_schema(_schema_generator: GenerateSchema, uuid_type: type[UUID]) -> cor
...
+                instance_schema,
+                core_schema.no_info_after_validator_function(path_validator, core_schema.str_schema()),
+            ],
+            custom_error_type='path_type',
+            custom_error_message='Input is not a valid path',
+            strict=True,
         ),
+        strict_schema=instance_schema,
+        serialization=core_schema.to_string_ser_schema(),
+        metadata=build_metadata_dict(js_override={'type': 'string', 'format': 'path'}),
```
Comment (hramezani) [inline pydantic/_internal/_std_types_schema.py:218]: I think `js_override` is not valid https://github.com/pydantic/pydantic/commit/c8b978ecd4136fa41baed58dee7d50cf6280cee5

Comment (samuelcolvin) [reply pydantic/_internal/_std_types_schema.py]: good point, fixed.

**Review by hramezani — APPROVED** [review]

**Comment by hramezani** [comment]
LGTM 👍

**Pull request merged**